client/app/vps/upgrade/vps-upgrade.html

```diff
@@ -0,0 +1,110 @@
+                                    </span>
+                                    <span class="col-xs-2 col-sm-2 py-0">
+                                        {{ upgrade.ram.value }} {{ 'unit_size_' + upgrade.ram.unit | translate }}
+                                    </span>
+                                    <span class="col-xs-3 col-sm-2 py-0">
+                                        {{ upgrade.disk.value }} {{ 'unit_size_' + upgrade.disk.unit | translate }}
+                                    </span>
+                                    <span class="col-xs-4 py-0">
+                                        <span data-ng-if="upgrade.isAvailable && !upgrade.isCurrentModel"
+                                              class="red bold">
+                                            {{ upgrade.monthPrice }} {{ 'vps_configuration_upgradevps_step1_table_price_monthly' | translate }}
+                                        </span>
+                                        <span data-ng-if="upgrade.isCurrentModel"
+                                              class="label label-info"
+                                              data-translate="vps_configuration_upgradevps_current"></span>
+                                    </span>
+                                </span>
+                            </label>
+                        </div>
+                        <div>
+                            <div>
+                                {{ 'vps_configuration_upgradevps_step1_warning' | translate:'{ monthPrice: $ctrl.getCurrentModel().monthPrice }' }}
+                            </div>
+                            <small class="mt-4 font-italic" data-translate="vps_configuration_upgradevps_step1_warning2"></small>
+                        </div>
+                    </cui-wizard-step>
+
+                    <cui-wizard-step data-step-title="{{ 'vps_configuration_upgradevps_step2_title' | translate }}"
+                                     data-step-on-init="$ctrl.initVpsConditions()"
+                                     data-step-loaded="!$ctrl.loaders.step2 && $ctrl.order"
+                                     data-step-disabled="$ctrl.step < 2"
+                                     data-step-completed="$ctrl.agree.value"
+                                     data-step-on-complete="$ctrl.loadNextStep(3)">
+                            <div data-contracts="$ctrl.order.contracts"
+                                 data-ng-model="$ctrl.agree.value"
+                                 data-contracts-validated
+                                 data-full-text="false"></div>
+                    </cui-wizard-step>
+
+                    <cui-wizard-step data-step-title="{{ 'vps_configuration_upgradevps_step3_title' | translate }}"
+                                     data-step-on-init="$ctrl.initVpsConditions()"
+                                     data-step-loaded="!$ctrl.step > 2 && $ctrl.order"
```

Comment:
We shouldn't use index for states since it doesn't allow for dynamic step injection if needed.
So step should be named (with an unique code).  It's not a full blown state machines but it's a good compromise ;)

Reply:
Agreed, we are thinking of evolving the process used for steps chaining